service/s3/s3crypto/mat_desc.go

```diff
@@ -0,0 +1,81 @@
+package s3crypto
+
+import (
+	"encoding/json"
+)
+
+// MaterialDescription is used for describing the materials
+// that are being encrypted.
+type MaterialDescription interface {
+	Set(string, string)
+	Get(string) (string, bool)
+	EncodeDescription() (string, error)
+	DecodeDescription([]byte) error
+	// TODO: Should we do this differently?
+	GetData() map[string]*string
```

Comment:
I originally tried it with type aliasing. The issue here is we can't cast a type aliased thing to a `map[string]*string`